js/exporter/jspdf/pdf_grid.js

```diff
@@ -48,17 +48,111 @@ export class PdfGrid {
         let currentTableIndex = 0;
         let currentTableCells = [];
         for(let cellIndex = 0; cellIndex < cells.length; cellIndex++) {
+            const currentCell = cells[cellIndex];
             const isNewTableColumn = this._splitByColumns.filter((splitByColumn) => splitByColumn.columnIndex === cellIndex)[0];
             if(isNewTableColumn) {
                 this._currentHorizontalTables[currentTableIndex].addRow(currentTableCells, rowHeight);
+                if(currentCell.colSpan > 0) {
+                    this._splitColSpanArea(cells, cellIndex);
+                }
                 currentTableIndex++;
                 currentTableCells = [];
             }
-            currentTableCells.push(cells[cellIndex]);
+            currentTableCells.push(currentCell);
         }
         this._currentHorizontalTables[currentTableIndex].addRow(currentTableCells, rowHeight);
     }
 
+    _splitColSpanArea(cells, splitIndex) {
+        const { startColSpanCellIndex, endColSpanCellIndex, colSpanCellText } = this._getColSpanArea(cells, splitIndex);
+
+        if(startColSpanCellIndex !== -1 && endColSpanCellIndex !== -1) {
+            const leftSpanCells = splitIndex - startColSpanCellIndex - 1;
+            for(let index = startColSpanCellIndex; index < splitIndex; index++) {
+                if(leftSpanCells > 0) {
+                    cells[index].colSpan = leftSpanCells;
+                } else {
+                    delete cells[index].colSpan;
+                }
+            }
+            cells[splitIndex].text = colSpanCellText;
+            const rightSpanCells = endColSpanCellIndex - splitIndex;
+            for(let index = splitIndex; index <= endColSpanCellIndex; index++) {
+                if(rightSpanCells > 0) {
+                    cells[index].colSpan = rightSpanCells;
+                } else {
+                    delete cells[index].colSpan;
+                }
+            }
+        }
+    }
+
+    _getColSpanArea(cells, cellIndex) {
```

Comment:
Let's change it to this:

    _findMergedRangeStartIndex(cells, targetCellIndex)
        let currentCellIndex = 0;
        while(currentCellIndex <= targetCellIndex; currentCellIndex++) {
            let currentColSpan = cells[currentCellIndex].colSpan;
            if(currentColSpan > 0) {
                if((currentCellIndex + currentColSpan) >= targetCellIndex) {
                    return currentCellIndex;
                } else {
                    currentCellIndex = currentMergeRange.endCellIndex + 1;
                }
            } else {
                currentCellIndex++;
            }
        }
    }

Reply:
OK